perfkitbenchmarker/linux_benchmarks/dpb_testdfsio_benchmark.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Perform Distributed i/o benchmark on data processing backends.
+This test writes into and then subsequently reads a specified number of
+files. File size is also specified as a parameter to the test.
+The benchmark implementation accepts list of arguments for both the above
+parameters and generates one sample for each cross product of the two
+parameter values. Each file is accessed in a separate map task.
+"""
+
+import copy
+import datetime
+
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import dpb_service
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker.dpb_service import BaseDpbService
+
+BENCHMARK_NAME = 'dpb_testdfsio_benchmark'
+
+BENCHMARK_CONFIG = """
+dpb_testdfsio_benchmark:
+  description: Run testdfsio on dataproc and emr
+  dpb_service:
+    service_type: dataproc
+    worker_group:
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-4
+          boot_disk_size: 1500
+        AWS:
+          machine_type: m4.xlarge
+      disk_spec:
+        GCP:
+          disk_type: nodisk
+        AWS:
+          disk_size: 1500
+          disk_type: gp2
```

Comment:
I cannot speak to other resources but for dpb in particular, the difference in how the worker node disk (type and size) is specified for the different clouds is due to the difference in their cli specifications. Dataproc allows direct specification of boot disks which the workers use for local computation while EMR requires EBS specification attached to worker's instance group.